src/provider/model/FinEnvironment.ts

```diff
@@ -77,7 +77,8 @@ export class FinEnvironment extends AsyncInit implements Environment {
         identity = parseIdentity(identity);
         const id = getId(identity);
 
-        const {creationTime, index} = this._seenWindows[id];
+        const seenWindow = this._seenWindows[id] || {creationTime: 0, index: this._windowsCreated++};
```

Comment:
Would comment need for this. And would it make more sense for creation time to be time of connection?